detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunction.kt

```diff
@@ -28,7 +28,7 @@ class UndocumentedPublicFunction(config: Config = Config.empty) : Rule(config) {
     )
 
     override fun visitNamedFunction(function: KtNamedFunction) {
-        if (function.funKeyword == null && function.isLocal) return
+        if (function.isLocal) return
 
         if (function.docComment == null && function.shouldBeDocumented()) {
```

Comment:
Maybe this if clauses should be merged for even more clarity.

```suggestion
        if (!function.isLocal && function.docComment == null && function.shouldBeDocumented()) {
```